src/MissionManager/LandingComplexItem.cc

```diff
@@ -415,6 +436,27 @@ bool LandingComplexItem::_scanForItem(QmlObjectListModel* visualItems, bool flyV
         scanIndex += CameraSection::stopTakingPhotosCommandCount();
     }
 
+    scanIndex--;
+    bool useDoChangeSpeed = false;
+    double finalApproachSpeed = 0;
+    if (scanIndex >= 0 && scanIndex < visualItems->count()) {
+        item = visualItems->value<SimpleMissionItem*>(scanIndex);
+        if (item) {
+            MissionItem& missionItemChangeSpeed = item->missionItem();
+            if (missionItemChangeSpeed.command() == MAV_CMD_DO_CHANGE_SPEED &&
+                missionItemChangeSpeed.param1() >= 0 && missionItemChangeSpeed.param1() <= 3 &&
```

Comment:
> But that would mean that a mission sequence with a manually configured DO_CHANGE_SPEED item wouldn't draw a glide slope and such.

It wouldn't be a manually configured change speed, it would be a fully manually configured landing pattern. The change speed is after the DO_LAND_START. If you use the pattern, there is not way to manually get something in the middle of those items.

You are slurping a change speed command off the stack which you can't show the correct ui for in the pattern editor since the pattern editor only supports airspeed. The user won't really know what is happening.